wire-ios-data-model/Source/ManagedObjectContext/Migration/118-119/FixDuplicateOneOnOneConversationsAction.swift

```diff
@@ -0,0 +1,128 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+final class FixDuplicateOneOnOneConversationsAction: CoreDataMigrationAction {
+
+    override func execute(in context: NSManagedObjectContext) throws {
+
+        // 1) fetch all conversations to fix
+        guard let fake1On1Conversations: [ZMConversation] = try? fakeOneOnOneConversations(context: context), !fake1On1Conversations.isEmpty else {
+            return
+        }
+
+        let fakeOneOnOnePerUser = conversationsPerUser(from: fake1On1Conversations)
```

Comment:
Elsewhere in our code we often name dictionaries as "valueByKey", so in this case `conversationsByUser`.

Reply:
done